src/inference/__tests__/rerank.test.ts

```diff
@@ -0,0 +1,115 @@
+import { Inference } from '../inference';
+import type { PineconeConfiguration } from '../../data';
+import { inferenceOperationsBuilder } from '../inferenceOperationsBuilder';
+import { PineconeArgumentError } from '../../errors';
+import { RerankResult } from '../../pinecone-generated-ts-fetch/inference';
+
+let inference: Inference;
+const rerankModel = 'test-model';
+const myQuery = 'test-query';
+
+beforeAll(() => {
+  const config: PineconeConfiguration = { apiKey: 'test-api-key' };
+  const infApi = inferenceOperationsBuilder(config);
+  inference = new Inference(infApi);
+});
+
+test('Confirm throws error if no documents are passed', async () => {
+  const expectedError = new PineconeArgumentError(
+    'You must pass at least one document to rerank'
+  );
```

Comment:
nit: Personal preference, but I'd maybe just leave the expected error inline on the `expect().toEqual()`. Feels easier to parse.

Reply:
updated!